tests/integration-tests/conftest.py

```diff
@@ -363,24 +368,44 @@ def _config_renderer(config_file="pcluster.config.ini", **kwargs):
         env = Environment(loader=file_loader)
         rendered_template = env.get_template(config_file).render(**{**kwargs, **default_values})
         config_file_path.write_text(rendered_template)
-        add_custom_packages_configs(config_file_path, request, region)
-        _enable_sanity_check_if_unset(config_file_path)
+        add_custom_packages_configs(config_file_path, request)
         return config_file_path
 
     return _config_renderer
 
 
-def add_custom_packages_configs(cluster_config, request, region):
-    config = configparser.ConfigParser()
-    config.read(cluster_config)
-    cluster_template = "cluster {0}".format(config.get("global", "cluster_template", fallback="default"))
+def add_custom_packages_configs(cluster_config, request):
+    with open(cluster_config) as conf_file:
+        config_content = yaml.load(conf_file, Loader=yaml.SafeLoader)
+    if request.config.getoption("custom_chef_cookbook"):
+        if config_content.get("DevSettings") is None:
+            config_content["DevSettings"] = {}
+        if config_content["DevSettings"].get("Cookbook") is None:
+            config_content["DevSettings"]["Cookbook"] = {}
+        if config_content["DevSettings"]["Cookbook"].get("ChefCookbook") is None:
+            config_content["DevSettings"]["Cookbook"]["ChefCookbook"] = request.config.getoption("custom_chef_cookbook")
+
+    if config_content["Image"].get("CustomAmi") is None:
+        additional_filters = []
+        if request.config.getoption("createami_cookbook_git_ref"):
+            additional_filters.append(
+                {
+                    "Name": "tag:parallelcluster_cookbook_ref",
+                    "Values": [request.config.getoption("createami_cookbook_git_ref")],
+                }
+            )
+        config_content["Image"]["CustomAmi"] = retrieve_pcluster_ami_with_additional_filters(
+            config_content["Image"].get("Os"),
+            architecture=get_architecture_supported_by_instance_type(config_content["HeadNode"]["InstanceType"]),
+            additional_filters=additional_filters,
+        )
 
+    with open(cluster_config, "w") as conf_file:
+        yaml.dump(config_content, conf_file)
+
+
+"""
     for custom_option in [
```

Comment:
what about the rest of the method? do you plan to fix it in a separate PR?

Reply:
there are some cases where we pass to the `test_runner` global `pre-install` or `post-install` through the --pre-install or --post-install cli argument so we need to preserve this behavior. Let's follow-up on this.